snuba/environment.py

```diff
@@ -70,6 +70,12 @@ def setup_logging(level: Optional[str] = None) -> None:
     )
 
 
+def _traces_sampler(sampling_context: dict[str, Any]) -> int:
+    if sampling_context["transaction_context"]["op"] == "snuba_init":
+        return 1
+    return settings.SENTRY_TRACE_SAMPLE_RATE
+
```

Comment:
I don't think this works as you would want. 
https://github.com/getsentry/sentry-python/blob/c05bcf598c5455a6f35eabd18c840c4544c9392c/sentry_sdk/tracing.py#L723-L739

If you do not define a `trace_sampler`, snuba honors the decision made by sentry, thus making the trace consistent. Which is what we want. 
But if you define a `traces_sampler`, your sampler needs to explicitly honor the parent decision, you cannot just return a sampling rate.


Also the call `start_transaction` is supposed to have a way to force a the sampling 
```
start_transaction(name: "my transaction", sampled: True)
```
so you should not need this,

Reply:
as per the conversation that happened in https://github.com/getsentry/snuba/pull/3018, my understanding is that we could just be increasing `SENTRY_TRACE_SAMPLE_RATE` in our prod settings, which would still honor the sentry decisions it would just mean new traces in snuba would get that new rate.

Reply:
You do not want to sample 100% of the transactions started in snuba that are not of the specific type Rahul is interested into. That would also sample the massive amount of transactions that seem to originate from our consumers that we never managed to investigate. (Looking at the outcomes table there are several orders of magnitude more transactions dropped by the sdk than those that reach sentry. These are likely to originate in our consumers).

Reply:
I thought there were no other `start_transaction` beside rahul's. But anyway according to our docs the `start_transaction` does have a way to enable sampling and it seems like it overrides whatever sample rate is else where so filippo's suggestion still stands

Reply:
I see, I'll try setting `sampled: True` and see if it'll force Sentry to sample the transaction. As far as Snuba transactions go, this is the only transaction explicitly started within Snuba but I didn't know whether the sampling rate in settings only mattered for transactions started within Snuba or all transactions Snuba's logic is a part of.

Reply:
There are some docs on the `traces_sampler` function [here](https://docs.sentry.io/platforms/python/guides/falcon/configuration/options/#traces-sampler). Filippo's link actually explains it better.

The sampling happens to every transaction started with our DSN. The transactions are still started in Snuba, even though they are implicitly created by our integrations. The web API creates transactions through the Flask integration. Because we don't have `traces_sampler` set, we are effectively sampling based on whether the parent transaction (from Sentry in our case) is sampling or not. So if Sentry samples, so does Snuba, and vice versa.